### ISSUE TITLE
borg config storage_quota returns invalid value for a value like 120G

### ISSUE THREAD

**Issue by mauromol** [borgbackup/borg#5639]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Bug

#### Your borg version (borg -V).

1.1.14

#### Operating system (distribution) and version.

Debian 8.11

#### Hardware / network configuration, and filesystems used.

ext4 on mdadm RAID set

#### How much data is handled by borg?

24,75 TB original, 12,50 TB compressed, 67,14 GB compressed.

#### Full borg commandline that lead to the problem (leave away excludes and passwords)

borg config /local/repository/path storage_quota 120G

## Describe the problem you're observing.

The command fails with an "invalid value" error. However I would expect the syntax to be valid.

I think the cause is at: https://github.com/borgbackup/borg/blob/699256edbd072c60b995c493c2146aabe27905bb/src/borg/archiver.py#L1744
It seems like it's expecting an integer literal. This is in contrast with, for instance, `additional_free_space` parameter:
https://github.com/borgbackup/borg/blob/699256edbd072c60b995c493c2146aabe27905bb/src/borg/archiver.py#L1753
However the documentation of borg init says that the quota amount can be specified with a syntax like 120G: https://borgbackup.readthedocs.io/en/stable/usage/init.html
Also, I think it's error prone to specify such a configuration parameter value using an integer literal.

#### Include any warning/errors/backtraces from the system logs

```
Local Exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/borg/archiver.py", line 4591, in main
    exit_code = archiver.run(args)
  File "/usr/local/lib/python3.4/dist-packages/borg/archiver.py", line 4523, in run
    return set_ec(func(args))
  File "/usr/local/lib/python3.4/dist-packages/borg/archiver.py", line 176, in wrapper
    return method(self, args, repository=repository, **kwargs)
  File "/usr/local/lib/python3.4/dist-packages/borg/archiver.py", line 1884, in do_config
    validate(section, name, args.value)
  File "/usr/local/lib/python3.4/dist-packages/borg/archiver.py", line 1795, in repo_validate
    raise ValueError('Invalid value') from None
ValueError: Invalid value

Platform: Linux openmediavault 3.16.0-11-686-pae #1 SMP Debian 3.16.84-1 (2020-06-09) i686
Linux: debian 8.11 
Borg: 1.1.14  Python: CPython 3.4.2 msgpack: 0.5.6
PID: 24510  CWD: /media/f6280d17-6fef-460e-bb67-dfe3c26d87bf/home/borg
sys.argv: ['/usr/local/bin/borg', 'config', '/media/f6280d17-6fef-460e-bb67-dfe3c26d87bf/backup/borg/hppb/mauro/home', 'storage_quota', '120G']
SSH_ORIGINAL_COMMAND: None
```

**Comment by ThomasWaldmann** [comment]
Yeah, looks like there is some inconsistency due to differently capable argument parsing.

If somebody cleans this up, please check all the config values.

**Comment by ThomasWaldmann** [comment]
Similar issue for `max_segment_size`.

**Comment by ThomasWaldmann** [comment]
Fixed by #5712 and #5707.
